core/src/main/scala/io/delta/tables/execution/DeltaTableOperations.scala

```diff
@@ -62,22 +77,36 @@ trait DeltaTableOperations extends AnalysisHelper { self: DeltaTable =>
   }
 
   protected def executeGenerate(tblIdentifier: String, mode: String): Unit = {
-    val tableId: TableIdentifier = sparkSession
-      .sessionState
-      .sqlParser
+    val tableId: TableIdentifier = sparkSession.sessionState.sqlParser
       .parseTableIdentifier(tblIdentifier)
     val generate = DeltaGenerateCommand(mode, tableId, self.deltaLog.options)
     toDataset(sparkSession, generate)
   }
 
   protected def executeUpdate(
       set: Map[String, Column],
-      condition: Option[Column]): Unit = improveUnsupportedOpError {
+      condition: Option[Column],
+      userMetadata: Option[String] = None): Unit = improveUnsupportedOpError {
+    val previousMetadata = sparkSession.conf.getOption(
+      "spark.databricks.delta.commitInfo.userMetadata")
+    userMetadata match {
+      case Some(metadata) => sparkSession.conf.set(
+        "spark.databricks.delta.commitInfo.userMetadata", metadata)
+      case None => None
+    }
     val assignments = set.map { case (targetColName, column) =>
       Assignment(UnresolvedAttribute.quotedString(targetColName), column.expr)
     }.toSeq
-    val update = UpdateTable(self.toDF.queryExecution.analyzed, assignments, condition.map(_.expr))
-    toDataset(sparkSession, update)
+    val update =
+      UpdateTable(self.toDF.queryExecution.analyzed, assignments, condition.map(_.expr))
+    val dataset = toDataset(sparkSession, update)
+    previousMetadata match {
+      case Some(metadata) => sparkSession.conf.set(
+        "spark.databricks.delta.commitInfo.userMetadata", metadata)
```

Comment:
what happens when there are concurrent delete operations that set different user metadata ?

Reply:
I believe that you would get non-deterministic behaviour and probably the wrong metadata written to one. The issue is that I believe delete and update operations happen through spark APIs that we can't pass userMetadata to (can we?) unlike the merge operation, that is all delta and does allow metadata to be scoped to the operation.

Reply:
@rahulsmahadev can you provide help here?